src/components/HotspotEditorModal/HotspotTooltipTab/HotspotTooltipTab.jsx

```diff
@@ -0,0 +1,221 @@
+      PropTypes.number,
+      PropTypes.bool,
+      colorPropType,
+    ])
+  ),
+  /** Internationalisation strings */
+  i18n: PropTypes.shape({
+    deleteButtonLabelText: PropTypes.string,
+    deleteButtonIconDescription: PropTypes.string,
+    titleInputLabelText: PropTypes.string,
+    titleInputPlaceholderText: PropTypes.string,
+    descriptionTextareaLabelText: PropTypes.string,
+    descriptionTextareaPlaceholderText: PropTypes.string,
+    iconDropdownLabelText: PropTypes.string,
+    colorDropdownLabelText: PropTypes.string,
+    colorDropdownTitleText: PropTypes.string,
+  }),
+  /** If present, the msg and an info icon will be the only thing showing  */
+  infoMessage: PropTypes.string,
+  /** Callback for when any of the form element's value changes */
+  onChange: PropTypes.func.isRequired,
+  /** Callback for when the delete button is clicked */
+  onDelete: PropTypes.func.isRequired,
+  /**
+   * Used to overide the internal components and props for the text based input
+   * if need should arise to manage defaultValue, length or validation etc.
+   */
+  overrides: PropTypes.shape({
+    titleTextInput: OverridePropTypes,
+    decriptionTextArea: OverridePropTypes,
+  }),
+  /** The id of the form elememt that should be initially focused */
+  primaryInputId: PropTypes.string,
+  /** Id that can be used for testing */
+  testID: PropTypes.string,
+};
+
+const defaultProps = {
+  hotspotIconFillColors: undefined,
+  hotspotIcons: [],
+  formValues: {},
+  i18n: {
+    deleteButtonLabelText: 'Delete hotspot',
+    deleteButtonIconDescription: 'Delete this hotspot',
+    titleInputLabelText: 'Title',
+    titleInputPlaceholderText: 'Enter title for the tooltip',
+    descriptionTextareaLabelText: 'Description',
+    descriptionTextareaPlaceholderText: 'Enter description for the tooltip',
+    iconDropdownLabelText: 'Select an Icon',
+    iconDropdownTitleText: 'Icon',
+  },
+  infoMessage: undefined,
+  overrides: undefined,
+  primaryInputId: undefined,
+  testID: 'HotspotTooltipTab',
+};
+
+const preventFormSubmission = (e) => e.preventDefault();
+
+const HotspotTooltipTab = ({
+  hotspotIcons,
+  hotspotIconFillColors,
+  formValues,
+  i18n,
+  infoMessage,
+  primaryInputId,
+  onChange,
+  onDelete,
+  overrides,
+  testID,
+}) => {
+  const {
+    deleteButtonLabelText,
+    deleteButtonIconDescription,
+    titleInputLabelText,
+    titleInputPlaceholderText,
+    descriptionTextareaLabelText,
+    descriptionTextareaPlaceholderText,
+    iconDropdownLabelText,
+    iconDropdownTitleText,
+    colorDropdownLabelText,
+    colorDropdownTitleText,
+  } = merge({}, defaultProps.i18n, i18n);
+
+  const currentIconColor = formValues?.color?.carbonColor ?? 'currentcolor';
+
+  const renderInfoMessage = () => (
+    <div className={`${iotPrefix}--hotspot-editor--tooltip-info-message`}>
+      <InformationFilled24 />
+      <p>{infoMessage}</p>
+    </div>
+  );
+
+  const renderColorIconContainer = () => (
```

Comment:
I think this is a matter of coding style and for me this is about readability and maintainability :-) 
I try to follow the clean code principles which calls for small functions* and I think it is a common pattern also to split JSX into smaller pieces when when it grows too large (either as new components or additional render functions). Another clean code principle I've learned to appreciate is to avoid writing comments if I can structure my code in a way that makes commenting redundant since comments always risk becoming irrelevant or incorrect when code changes.

That being said, the choice when to plit and not is a bit arbitrary and I'm not religious about these things so if you feel strongly about not splitting this I will move this code back into the JSX?  

*) Although these functions are not considered "clean" since they rely on variables (the props) which are not parameters.

Reply:
I agree with you on all points, it's a matter of taste for sure. My feedback comes from some [recent reading].(https://kentcdodds.com/blog/aha-programming#aha-) I think the codebase would benefit from moving to a mentality of optimizing for change first and waiting to abstract to ensure we make the right abstraction. One long block of jsx allows you to read straight down avoiding scroll bouncing up to the abstractions and then back down to the render function.

By no means a blocker to merging, just some discussion. We should continue to discuss in the future! 🙂

Reply:
Thanks for the link, I find myself being led to Kent C Dodds' blog more and more nowadays. I'm also happy to explore this further in the future.